networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/client_nsx.py

```diff
@@ -0,0 +1,206 @@
+                        return response
+
+                    last_err = "Error Code={} Message={}"\
+                        .format(response.status_code, response.content)
+                    log_msg = "Request={} Response={}".format(requestInfo,
+                                                              last_err)
+
+                    # Handle resource not found gently
+                    if is_not_found(response):
+                        LOG.info("Unable to find Resource={}"\
+                            .format(kwargs["path"]))
+                        LOG.debug(log_msg)
+                        return response
+
+                    if response.status_code in [401, 403]:
+                        self._login()
+                        continue
+
+                    # Retry for The object AtomicRequest/10844 is already present in the system.
+                    if not is_atomic_request_error(response):
+                        # skip retry on the ramaining NSX errors
+                        LOG.error(log_msg)
+                        break
+                except (HTTPError, ConnectionError, ConnectTimeout) as err:
+                    last_err = err
+                    LOG.error("Request={} Response={}".format(requestInfo,
+                                                              last_err))
+
+                msg = pattern.format(attempt, until, pause, method)
+
+                LOG.debug(msg)
+                eventlet.sleep(pause)
+
+            raise Exception(msg, last_err)
+
+        return decorator
+
+
+
+class Client:
+
+    def __init__(self):
+        rate=cfg.CONF.NSXV3.nsxv3_requests_per_second
+        timeout=cfg.CONF.NSXV3.nsxv3_requests_per_second_timeout
+        limit=1
+
+        self._api_scheduler = Scheduler(rate, limit, timeout)
+
+        self._timeout = cfg.CONF.NSXV3.nsxv3_request_timeout
+
+        self._base_path = 'https://{}:{}'.format(
+            cfg.CONF.NSXV3.nsxv3_login_hostname,
+            cfg.CONF.NSXV3.nsxv3_login_port
+        )
+
+        self._login_timestamp = 0
+        self._login_path = "/api/session/create"
+        self._login_data = {
+            "j_username": cfg.CONF.NSXV3.nsxv3_login_user,
+            "j_password": cfg.CONF.NSXV3.nsxv3_login_password
+        }
+
+        self._session = requests.session()
+
+        if cfg.CONF.NSXV3.nsxv3_suppress_ssl_wornings:
+            self._session.verify = False
+            requests.packages.urllib3.disable_warnings()
+
+        self._version = None
+
+    @property
+    def version(self, refresh=False):
+        if not self._version or refresh:
+            resp = self.get(path="/api/v1/node/version")
+            if resp.ok:
+                self._version = resp.json()['product_version']
+        return versionutils.convert_version_to_tuple(self._version)
+
+    def _login(self):
+        LOG.info("Session token - acquiring")
+        now = int(datetime.datetime.now().strftime("%s"))
```

Comment:
In case you ever want the same result but with shorter code:
```python
>>> datetime.datetime.now().strftime('%s')
'1620027230'
>>> time.time()
1620027232.8798525
>>> int(time.time())
1620027275
```